example_queries/envoy/get_service_name.rs.ref

```diff
@@ -36,7 +36,14 @@ pub fn collect_envoy_properties(
     let mut prop_tuple: Property;
     let mut prop_tuple_wrapped: Option<Property>;
     prop_tuple_wrapped = fetch_property(&http_headers.workload_name,
-                                        &"node.metadata.WORKLOAD_NAME".split(".").collect(),
+                                        &vec!["node", "metadata", "WORKLOAD_NAME", ],
+                                        http_headers);
+                                            if let Some(prop_tuple) = prop_tuple_wrapped {
+            fd.unassigned_properties.push(prop_tuple);
+            } else {
+                return Err(());
+            }prop_tuple_wrapped = fetch_property(&http_headers.workload_name,
```

Comment:
These duplicates happen because the properties are 'different' in the sense that you have `a` and `c` as parent nodes. I do not know what the implications are here but it may make sense to check equality of properties based on the members. But then one needs to be careful about constants or variables that do not have members.